### PR TITLE
Fix Angular "bug" whereby anchor tags don't work over slow connections

### PR DESCRIPTION
**Pull request by aeijdenberg**
Fix issue reported in https://github.com/cloudfoundry/cli-plugin-repo/pull/279#issuecomment-462459606

Fix is to remove Angular.

### CODE REVIEW

**Comment by aeijdenberg** [comment]
@willmurphyscode - this PR should fix the Angular issue.

File: main_test.go [hunk 1]
```diff
@@ -47,14 +49,15 @@ var _ = Describe("Integration", func() {
...
-				b, err := ioutil.ReadFile("ui/index.html")
-				Expect(err).NotTo(HaveOccurred())
-
 				defer response.Body.Close()
 				contents, err := ioutil.ReadAll(response.Body)
 				Expect(err).NotTo(HaveOccurred())
 
-				Expect(string(contents)).To(Equal(string(b)))
+				// sanity test that at least one thing is present
+				Expect(strings.Index(string(contents), "doctor scans your deployed applications") >= 0).To(BeTrue())
```
Comment (jenspinney) [inline main_test.go]:
```suggestion
				Expect(contents).To(ContainSubstring("doctor scans your deployed applications"))
```

File: main_test.go [hunk 2]
```diff
@@ -47,14 +49,15 @@ var _ = Describe("Integration", func() {
...
+
+				// and that the template finishes rendering without aborting due to an error
+				Expect(strings.Index(string(contents), "</html>") >= 0).To(BeTrue())
```
Comment (jenspinney) [inline main_test.go]:
```suggestion
				Expect(contents).To(ContainSubstring("</html>"))
```

**Comment by jenspinney** [comment]
Hi @aeijdenberg, thanks for this work! I left some suggestions for minor fixes to the test assertions, but overall I like the direction.

We're still discussing the general approach as a team, and I want to make sure I'm not alone on the team on liking it before I merge it. You can see my thoughts here if you're curious: https://www.pivotaltracker.com/story/show/163893342/comments/199531284

My big assumption here was that you removed `rata` because it was hard to hook in Go templating with it?

It'd be nice to have a little more description in the commit message itself. Yes, this change will technically fix the anchor tags, but the bigger point is that we're replacing Angular with Go templating and rendering the list on the server instead of in the browser. At the moment that feels just like a small footnote instead of the main point.

**Comment by aeijdenberg** [comment]
@jenspinney - yes removed Angular because it seemed easier to simply remove than add more Angular to restore functionality that basic HTML has supported for over 20 years!

Removed `rata` because I spent maybe an hour mucking around trying to get it to work with an additional route to handle the templating without luck. Given that this is essentially a single page app, it seems unnecessary to need to use another 3rd party framework to handle routing.

I also noticed that `components.css` is over 1MB - Chrome seems to think that about 95% of that code is unused, but I was unable to easily make a smaller version of that - but that might be an interesting project for someone else to take on if they wish to reduce page load time.

Up to your team if you merge it or look for other approaches. My feedback would be for a small app, and one that's maintained on a somewhat part-time basis (as implied by the call to help), then the less 3rd party libraries used the better - less code, less to patch, lower bar to entry for new contributors.

**Comment by jenspinney** [comment]
Makes sense, and I'll chat with the team tomorrow about it. Thanks again for the contribution!

**Comment by willmurphyscode** [comment]
@aeijdenberg Thank you so much for doing this extra work! It should be deployed momentarily.

**Pull request merged**